client/rest-high-level/src/main/java/org/elasticsearch/client/indices/AbstractDataStreamRequest.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client.indices;
+
+import org.elasticsearch.client.Validatable;
+
+public abstract class AbstractDataStreamRequest implements Validatable {
```

Comment:
I'm not sure this abstraction is buying us anything, we don't have a need anywhere to deal with a data stream request without caring what type it is, and it looks like none of the subclasses implement the validation `validate()` method.

Other than sharing a name they don't have much to tie themselves together, so I think we should remove this class.

Reply:
You're right, I've removed abstract class